File: tool/templates/template.cs
using System;
using System.Collections.Generic;


namespace Aoc
{
    class Solution
    {
        private static string solve(string input) {
            // Your code goes here
            return "Nothing";
        }

        public static void Main(string[] args) {
            string input = args[0];
            var watch = new System.Diagnostics.Stopwatch();
            watch.Start();
            string result = solve(input);
            watch.Stop();
            Console.WriteLine("_duration: " + watch.ElapsedMilliseconds + "\n" + result);
        }
        
    }
}

File: tool/templates/template.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"time"
)

func run(s string) interface{} {
	// Your code goes here
}

func main() {
	// Uncomment this line to disable garbage collection
	// debug.SetGCPercent(-1)

	argsWithProg := os.Args
	var input []byte
	var err error
	if len(argsWithProg) > 1 {
		input,err = ioutil.ReadFile(argsWithProg[1])
		if err != nil {
			panic(err)
		}
	} else {
		// Read input from stdin
		input, err = ioutil.ReadAll(os.Stdin)
		if err != nil {
			panic(err)
		}
	}

	// Start resolution
	start := time.Now()
	result := run(string(input))

	// Print result
	fmt.Printf("_duration:%f\n", time.Since(start).Seconds()*1000)
	fmt.Println(result)
}
